### DISCUSSION TITLE
v0.0.1.8-pw3 新功能投票

### DISCUSSION THREAD

**Discussion by Johnserf-Seed** [Johnserf-Seed/f2#203] [Polls]
`v0.0.1.7-pw2`的开发工作陆陆续续结束了，2024也即将到尾声了，尽管过程有些曲折，断更了一会，但在文档更新后就会发布了。

- 添加了通知app(感谢Bark)，可以在任务完成后推送通知
- 支持了更多的douyin直播间弹幕消息类型回调
- 支持了tiktok直播间弹幕采集
- 修复了weibo与x的功能
- 极大的优化了过滤器性能
- 优化了进度条
- 完整支持自定义ua的abogus参数
- 一大堆的底层修改
- 等等……

年底前会更新`v0.0.1.8-pw3`版本，已经确认会加入`bilibili`与`网易云音乐`这2个app，该投票是作为新功能的补充。

**Comment by CCMonkeyss** [comment]
辛苦了~

有个建议，抖音下载的文件有时候会下载失败，但是从文件名看不出来，
文件大小有的是0k，有的很大（但是没下载完整），
能不能给未下载完的文件加个标记，方便一次删除或是续传。

**Reply by Johnserf-Seed** [reply]
可以给0kb的文件也用.tmp来命名，这样子不需要删除就可以续传了

**Reply by CCMonkeyss** [reply]
> 可以给0kb的文件也用.tmp来命名，这样子不需要删除就可以续传了

感谢回复🙏，0kb文件其实还好处理，
我的情况是会有一些未下载完的文件，但是文件名是正常的，如果不点开检查就看不出来。
比如有个“123.mp4”的2MB文件，没下载完可能就是“123.mp4”  1.1MB的文件，
点开才能看出来视频内容少了后面一段。

因为我是直接下载自己的like列表，所以文件夹里文件很多，
如果此类未下载完成的文件有不同的副档名或是特殊标记，
就能很方便的一次性清理再重新下载，当然如果能续传就更好了。

还有，如果文件夹中有要下载的同名文件，下载会直接跳过，
所以上面说的未完成文件就不会被修复。

**Reply by Johnserf-Seed** [reply]
> > 可以给0kb的文件也用.tmp来命名，这样子不需要删除就可以续传了
> 
> 
> 
> 感谢回复🙏，0kb文件其实还好处理，
> 
> 我的情况是会有一些未下载完的文件，但是文件名是正常的，如果不点开检查就看不出来。
> 
> 比如有个“123.mp4”的2MB文件，没下载完可能就是“123.mp4”  1.1MB的文件，
> 
> 点开才能看出来视频内容少了后面一段。
> 
> 
> 
> 因为我是直接下载自己的like列表，所以文件夹里文件很多，
> 
> 如果此类未下载完成的文件有不同的副档名或是特殊标记，
> 
> 就能很方便的一次性清理再重新下载，当然如果能续传就更好了。
> 
> 
> 
> 还有，如果文件夹中有要下载的同名文件，下载会直接跳过，
> 
> 所以上面说的未完成文件就不会被修复。

这种情况比较有意思，我可能需要特殊手段来复现一下，下个版本会修复你提到的这个问题🙋 当初不用数据库来保存下载记录的一点就是删除的时候比较麻烦。 新版我可能要做一些改动来搞定这个问题

**Reply by zhiben5201** [reply]
只要报错了基本就是这位老哥说的情况，开始我还想甄别筛选一下，后来实在是没精力放弃了。下载图片时是会出现0KB的情况，下载视频下载不完整的就真没能力筛选了

**Reply by CCMonkeyss** [reply]
> > > 可以给0kb的文件也用.tmp来命名，这样子不需要删除就可以续传了
> > 
> > 
> > 感谢回复🙏，0kb文件其实还好处理，
> > 我的情况是会有一些未下载完的文件，但是文件名是正常的，如果不点开检查就看不出来。
> > 比如有个“123.mp4”的2MB文件，没下载完可能就是“123.mp4”  1.1MB的文件，
> > 点开才能看出来视频内容少了后面一段。
> > 因为我是直接下载自己的like列表，所以文件夹里文件很多，
> > 如果此类未下载完成的文件有不同的副档名或是特殊标记，
> > 就能很方便的一次性清理再重新下载，当然如果能续传就更好了。
> > 还有，如果文件夹中有要下载的同名文件，下载会直接跳过，
> > 所以上面说的未完成文件就不会被修复。
> 
> 这种情况比较有意思，我可能需要特殊手段来复现一下，下个版本会修复你提到的这个问题🙋 当初不用数据库来保存下载记录的一点就是删除的时候比较麻烦。 新版我可能要做一些改动来搞定这个问题

正好复现了，截图供您参考

<img width="1280" alt="截屏2025-01-07 06 39 551" src="https://github.com/user-attachments/assets/e7946249-d561-4cff-94ba-315218a54811" />

<img width="1095" alt="截屏2025-01-07 06 42 08" src="https://github.com/user-attachments/assets/4cb7fa92-e500-4ecb-9c27-6d785f0245e7" />


报错后下载的文件如上图，只有377KB，删除了重新下载，完整文件是3768KB

**Reply by zhiben5201** [reply]
> > > > 可以给0kb的文件也用.tmp来命名，这样子不需要删除就可以续传了
> > > 
> > > 
> > > 感谢回复🙏，0kb文件其实还好处理，
> > > 我的情况是会有一些未下载完的文件，但是文件名是正常的，如果不点开检查就看不出来。
> > > 比如有个“123.mp4”的2MB文件，没下载完可能就是“123.mp4”  1.1MB的文件，
> > > 点开才能看出来视频内容少了后面一段。
> > > 因为我是直接下载自己的like列表，所以文件夹里文件很多，
> > > 如果此类未下载完成的文件有不同的副档名或是特殊标记，
> > > 就能很方便的一次性清理再重新下载，当然如果能续传就更好了。
> > > 还有，如果文件夹中有要下载的同名文件，下载会直接跳过，
> > > 所以上面说的未完成文件就不会被修复。
> > 
> > 
> > 这种情况比较有意思，我可能需要特殊手段来复现一下，下个版本会修复你提到的这个问题🙋 当初不用数据库来保存下载记录的一点就是删除的时候比较麻烦。 新版我可能要做一些改动来搞定这个问题
> 
> 正好复现了，截图供您参考
> 
> <img alt="截屏2025-01-07 06 39 551" width="1280" src="https://private-user-images.githubusercontent.com/126976235/400579935-e7946249-d561-4cff-94ba-315218a54811.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMTI1MTMsIm5iZiI6MTczNjIxMjIxMywicGF0aCI6Ii8xMjY5NzYyMzUvNDAwNTc5OTM1LWU3OTQ2MjQ5LWQ1NjEtNGNmZi05NGJhLTMxNTIxOGE1NDgxMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwN1QwMTEwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTA3YjU2MDE3ZTVkNTBhNDc0NDcyNDE2OGI2OGJiYjRhNGIyMGIyZmUxZDk4MjRkNjBkM2NjOTI4MjkyNmM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.zo2-_iddx4N-gRjyCipuE3AUfe_9vyC2rcd1RjHxxfY"> <img alt="截屏2025-01-07 06 42 08" width="1095" src="https://private-user-images.githubusercontent.com/126976235/400579987-4cb7fa92-e500-4ecb-9c27-6d785f0245e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMTI1MTMsIm5iZiI6MTczNjIxMjIxMywicGF0aCI6Ii8xMjY5NzYyMzUvNDAwNTc5OTg3LTRjYjdmYTkyLWU1MDAtNGVjYi05YzI3LTZkNzg1ZjAyNDVlNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwN1QwMTEwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMWUxOWYzNWMxZGIxYjdiYWFlNzA5ODc2NjgxY2Q4YmRmNjlhMDM0ZDg2NzE5ZjE0MmJlMGJkYzY2YTMyMWRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TlMUsuUOuVmo8JjOQYghLM48HCfstwHDSf5O3Z-MMhM">
> 报错后下载的文件如上图，只有377KB，删除了重新下载，完整文件是3768KB

你真的好厉害，还能找到报错的文件，我就根本找不到了

**Reply by Johnserf-Seed** [reply]
这个问题的根本原因在于抖音服务器返回的块大小未严格遵守 HTTP 标准，而不是我们的下载逻辑存在问题。在 httpx 中同样存在类似的情况（参考：[httpx#1927](https://github.com/encode/httpx/issues/1927)）。这个我在源码中特别注释了。

目前，这个问题已经在直播流下载模块中进行了特别处理，因此在直播下载中不会再出现某些片段丢失的情况。

接下来，我会在文件下载模块中也引入相同的处理逻辑，并加入文件大小验证机制。对于下载不完全的文件，将保留 .tmp 后缀，以避免不完整文件覆盖目标文件，也可以帮助筛选出下载出错的任务。

**Comment by feixiang-1** [comment]
```
https://www.douyin.com/note/7367212847862009126
```
这种提示:非实况图集，跳过实况下载
能保存这种笔记吗

**Reply by Johnserf-Seed** [reply]
可以的，只是现在抖音又无法在网页端查看实况笔记了，只要抖音后台恢复了就可以保存，代码是ok的

**Comment by lgcde** [comment]
能不能记录已下载的作品id，下次再下载的时候跳过
因为帐号昵称改变的时候就会重新建个文件夹，又把已下载的下载一遍，以及文件命名方式改变的时候也会重新下载一遍
![image](https://github.com/user-attachments/assets/508bc5e8-ee9e-42da-89b1-15448db20057)
![image](https://github.com/user-attachments/assets/c7ef1422-9236-4df7-bf8f-1bc849984dfe)

**Comment by Josoen** [comment]
辛苦了大佬，想问问小红书有没有支持的计划

**Reply by NormanMises** [reply]
同问

**Comment by sky8886** [comment]
你好，请问有没有关注的博主发新视频了，能提醒或一键自动下载更新的视频？

**Comment by yingtianhao** [comment]
大大需要小红书！
